lib/std/zig/ErrorBundle.zig

```diff
@@ -433,7 +433,12 @@ pub const Wip = struct {
         // The ensureUnusedCapacity call above guarantees this.
         const notes_start = wip.reserveNotes(@intCast(other_list.len)) catch unreachable;
         for (notes_start.., other_list) |note, message| {
-            wip.extra.items[note] = @intFromEnum(wip.addOtherMessage(other, message) catch unreachable);
+            // It's important that this does not directly assign to wip.extra.items[note]
+            // because addOtherMessage can potentially resize wip.extra. With a direct
+            // assignment, RLS means that it would attempt to assign into the stale slice
+            // in the case of a resize.
+            const note_index = @intFromEnum(wip.addOtherMessage(other, message) catch unreachable);
+            wip.extra.items[note] = note_index;
```

Comment:
I'm not entirely sure this is RLS related and am also unsure if there's an existing issue for this particular thing.

Semi-minimal example of the problem:

```zig
const std = @import("std");

test "footgun" {
    var list = std.ArrayList(usize).init(std.testing.allocator);
    defer list.deinit();

    list.items[0] = try reserve(&list);

    try std.testing.expectEqual(0, list.items[0]);
}

fn reserve(list: *std.ArrayList(usize)) !usize {
    try list.ensureUnusedCapacity(1);
    return 5;
}
```

This fails with `panic: index out of bounds: index 0, len 0` since it's trying to write into the empty `&[_]T{}` slice instead of the slice that was allocated during the call to `reserve`.

Reply:
This is working as designed. The intended usage is `unusedCapacitySlice()` or `addOne()`.